libs/p2p-crypto/src/test/kotlin/net/corda/p2p/crypto/protocol/api/AuthenticationProtocolInitiatorTest.kt

```diff
@@ -67,7 +67,7 @@ AQIDAQAB
             assertThat(avro.protocolCommonDetails.sessionId).isEqualTo("sessionId")
             assertThat(avro.supportedModes).containsOnly(ProtocolMode.AUTHENTICATION_ONLY)
             assertThat(avro.groupId).isEqualTo("group")
-            assertThat(avro.ourPublicKey.trim()).isEqualTo(publicKeyPem.trim())
+            assertThat(avro.ourPublicKey.replace("\n", System.lineSeparator()).trim()).isEqualTo(publicKeyPem.trim())
```

Comment:
Another option here is to convert the returned value to a `PublicKey` object so we can compare them cryptographically rather than doing a string comparison.
```suggestion
            assertThat(avro.ourPublicKey.toPublicKey()).isEqualTo(publicKey)
```
To do that, we can add a helper function to the bottom of this class:
```
private fun String.toPublicKey() : PublicKey =
        publicKeyFactory(this.reader())!!
```

Reply:
Added in commit #136720b